recipe/bld.bat

```diff
@@ -6,22 +6,32 @@ if errorlevel 1 exit 1
 set MENU_DIR=%PREFIX%\Menu
 IF NOT EXIST (%MENU_DIR%) mkdir %MENU_DIR%
 
+rem  Copy Spyder's icon
 copy %RECIPE_DIR%\spyder.ico %MENU_DIR%\spyder.ico
 
+rem  Replace variables in menu files
 for /F "delims=. tokens=1" %%i in ("%PKG_VERSION%") do set PKG_MAJOR_VER=%%i
+call :replace spyder-menu.json spyder-menu.json
```

Comment:
Is this replacement correct? Shouldn't it be replacing a `.bak` file, like in the line just below this one?

Reply:
`:replace` reads the lines in the first file and replaces the `__PKG_VERSION__` tokens with the major version number, then writes the lines to the second file. The default json file is for `menuinst` v2, so we just write back to the same file. Any `*.json` file will be executed by `menuinst`, so on the line below we also need to rename the `spyder-menu-v1.json` file so that `menuinst` does not try to execute on it. The `post-link.bat` script will determine whether v1 or v2 is in the base environment, and swap them if needed.

But perhaps it would be clearer to just have both files `spyder-menu-v1.json` and `spyder-menu-v2.json`, then let `post-link.bat` add `.bak` to the unnecessary file.

Reply:
Actually, the file names in the `Menu` directory should not change on install so that they will all be removed on uninstall. So it will be best to have `spyder-menu-v1.json.bak` and `spyder-menu.json`. Then copy v1 to `spyder-menu.json` (overwriting) only if necessary.